### ISSUE TITLE
QGIS 3.26.x - default value not updated as expected

### ISSUE THREAD

**Issue by aliasatp** [qgis/QGIS#50220]
### What is the bug or the crash?

Hi all, using QGIS 3.26.x releases I have an issue on a field set as default value with an expression. My expression is a "attribute()..." as a "JOIN" function which gets a value from a table built with a SQL. This table changes its values based on values of the editing layer. The field is updated only once, with the first change in the edition layer, then it remains populated with the same value even if the preview is right. In prevoius releases it worked only after saving the edition layer, not live. But it updated the value. 

### Steps to reproduce the issue

1. I have edition layer and a table created with SQL and based on edition layer fields
2. in edition layer add expression in default value using "attribuite()..." as a JOIN between edition layer and 'SQL' table
3. check  'Apply default value on update' box
4. change value in edition layer
5. first change I realised right behaviour
6. othes changes nothing happens but preview is right

### Versions

QGIS 3.26.1
QGIS 2.26.2

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by aliasatp** [comment]
I update this issue. QGIS 3.22.11 has the same bahaviour

**Comment by aliasatp** [comment]
Hi, I try to explain better the issue (or changement?) I realised. Until 3.22.8 (9 ?) I used default value and apply on update with an expression which on change get values from a related table or in general another table.
An example of expression:
case
when	 (array_to_string( relation_aggregate('RELATED_TAB', 'array_agg', "VALUE"))*"parametro")> 0 then 'CORRETTO'
when	 (array_to_string( relation_aggregate('RELATED_TAB', 'array_agg', "VALUE"))*"parametro")= 0 then 'FALSO'
ELSE
'VERIFICA'
END
Up to 3.22.8 the behaviour was as I expected, when I saved the feature (new or updated) the field get the correct value.
Now the behaviour is different. When I create the feature the field get a value, in this case "VERIFICA" and even if I change some other fields values (ex field: "parametro") it doesn't change anymore the field as expected although I save the feature. Updating the feature it seems to work but I am not so sure as I would and it was until 3.22.8
